### ISSUE TITLE
ArrayList: GetRange method: Debugger reports zero elements but accessing elements gives correct results.

### ISSUE THREAD

**Issue by FullValueRider** [WaynePhillipsEA/twinbasic#362]
**Describe the bug**
When examining the output of the GetRange method of an ArrayList, the twinBasic debugger reports that there are zero members BUT operations on the 'got' range work correctly.

```
Sub TestGetRangeOutput()
        
        Dim myArrayList As ArrayList
        Set myArrayList = New ArrayList
        
        With myArrayList
        
            .add 10
            .add 20
            .add 30
            .add 40
            .add 50
            .add 60
            
        End With
        
        Dim myList As ArrayList
        Set myList = myArrayList.getrange(0&, 3&)
        Debug.Print "Expecting 20", myList.item(1)
        Debug.Print "Expecting 3", myList.count
    
    End Sub
```

gives the expected output
```
Executing 'VBAlib.Pootle.TestGetRangeOutput'...
Expecting 20    20 
Expecting 3     3 
(time taken: 0.0002096s)
```

but when inspecting mylist at the first debug statemetn gives
```
{object: 366857708}
{no members}
```

**Desktop (please complete the following information):**
 - OS: Windows 10
 - twinBASIC compiler version [v0.10.4563]

**Comment by FullValueRider** [comment]
Further information

If you subsequently try to iterate over myList with a for each loop you get the error below

![image](https://user-images.githubusercontent.com/81257042/129869973-49f0f56e-0efa-4978-8279-24253e6eae7b.png)

**Comment by WaynePhillipsEA** [comment]
Technically, this is a bug in .NET, not tB.    The ArrayList obect that is returned from the GetRange method is not responding to requests for the IDispatch interface via IUnknown::QueryInterface, which we can prove here:

```
Dim pDisp As Object
Dim pUnk As stdole.IUnknown
Set pUnk = myList           ' first cast to IUnknown so that the interface inheritance type information for ArrayList is ignored by the compiler in the next call...
Set pDisp = pUnk            ' this should work, but instead the object returns E_NOINTERFACE to QueryInterface(IID_IDispatch)
```

This issue is avoided in VBx because it skips the request for the IDispatch interface since it knows from the early-bound type information that the ArrayList class/interface inherits from IDispatch.  The same thing happens in tB's For-Each implementation.   Its's a very subtle difference in handling between tB and VBx. It is not ordinarily a problem, but it needs to be changed to match the  VBx handling to better cope with this bad .NET implementation.

**Comment by DaveInCaz** [comment]
@WaynePhillipsEA  is that a previously known .NET issue? Or did you discover it looking into this?

**Comment by WaynePhillipsEA** [comment]
Discovered by debugging the tb debugger :)
Sorry, I have no idea if it is a generally known issue.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.4755, thanks for reporting
